steel/src/host/mod.rs

```diff
@@ -60,20 +63,11 @@ where
 {
     /// Creates a new provable [EvmEnv] from an alloy [Provider].
     pub async fn from_provider(provider: P, number: BlockNumberOrTag) -> Result<Self> {
```

Comment:
Should this now redundant function be removed, or at least marked as `#[deprecated]`?